src/core/mount.c

```diff
@@ -1712,8 +1720,11 @@ static int mount_setup_unit(
                 return log_warning_errno(r, "Failed to set up mount unit for '%s': %m", where);
 
         /* If the mount changed properties or state, let's notify our clients */
-        if (flags & (MOUNT_PROC_JUST_CHANGED|MOUNT_PROC_JUST_MOUNTED))
+        if (flags & (MOUNT_PROC_JUST_CHANGED|MOUNT_PROC_JUST_MOUNTED)) {
+                if (flags & MOUNT_PROC_DEVICE_CHANGED)
+                        device_found_node(m, what, DEVICE_FOUND_MOUNT, DEVICE_FOUND_MOUNT);
```

Comment:
in order to keep our indentations low I'd move these two lines before the outer if check here. i mean, you check the condition anyway explicitly, so ther's no reason to nest the two if expressions

Reply:
Ok, I'll do some more investigation.
I did do this at one point but found it less effective.
That is curious though because it's counter to what's expected in that the mount device should rarely change and so shouldn't make a difference.